middleware/cache.js

```diff
@@ -0,0 +1,36 @@
+const config = require('config')
+const redis = require('redis')
+const { isInCache, formatCacheData, clearOutdatedCache } = require('../services/cache')
+
+const cacheDataClient = redis.createClient(config.redis)
+cacheDataClient.connect()
+
+const saveToCache = async (url, data) => {
+  await cacheDataClient.set(url, formatCacheData(url, Date.now(), data))
+  await cacheDataClient.expire(url, config.redis.cacheDuration)
```

Comment:
Where is the mapping we talked about? you assume that across the entire app the url for updating and deleting is exactly the save as the get?